data/blog/wolvctf-2023/wannaflag.mdx

```diff
@@ -0,0 +1,335 @@
+  class="button leftmar "
+  >List of Flags</a
+>
+<p>It’s encrypted so you still need the password duhhh</p>
+```
+
+![GitHub](/static/images/wolvctf-2023/github.png)
+
+There happens to be an extraordinarily convenient `password-repo` repository on this GitHub account, which reveals a `passwords_stub.lst` file:
+
+![Password Repo](/static/images/wolvctf-2023/password-repo.png)
+
+Unfortunately, none of the passwords in this list were accepted by the spreadsheet. However, it reveals extraordinarily useful information: the password list is a stub (as hinted by part of the filename), and the real password of the spreadsheet was likely part of the original document.
+
+Let’s further analyze the passwords which were given to us. We have the name of a U.S. city/town, followed by a number of some kind. Googling the city followed by the number reveals that the number is actually the population of the city which proceeds it; this means that `Concord_city` has a population of `105186`, `Baton_Rouge_city` has a population of `225128`, etc. We can use this information to our advantage to reassemble the password list, assuming it is every U.S. city/town followed by its population.
+
+Upon further research, the population of the city was discovered to be sourced from the U.S. 2020 Census. [Census.gov](https://www.census.gov/data/tables/time-series/demo/popest/2020s-total-cities-and-towns.html), their official website, has various datasets of “Annual Estimates of the Resident Population for Incorporated Places of 50,000 or More.” Importing the data into Google Sheets, we can see that this was the exact dataset which was used to generate the stub:
+
+![Census](/static/images/wolvctf-2023/census.png)
+
+**Note**: See how the term “city,” “munincipality,” and “village” were lowercase in `password_stub.lst`? This is because the census uses these terms to categorize urban areas in the United States, and are not actually part of the area’s name. That is also why there are some entries which have two “cities” in their name (e.g. `Oklahoma City city`), as city is both part of the official city name and categorization.
+
+Let’s create a wordlist from this data. Although we can export it as a `.csv` and run a Python script to filter out the passwords, we can simply create a spreadsheet formula on another column to create the wordlist:
+
+```py
+=SUBSTITUTE(SUBSTITUTE(LEFT(B5,FIND(",",B5)-1)," ","_"),",","")&"_"&SUBSTITUTE(C5,",","")
+```
+
+![Wordlist](/static/images/wolvctf-2023/wordlist.png)
+
+We can now copy the column into a text file, `wordlist.txt`, and use it to crack the spreadsheet. We’ll be utilizing the `john` tool to achieve this.
+
+Firstly, we need the hash of the spreadsheet. We can use the [`office2john.py`](https://github.com/openwall/john/blob/bleeding-jumbo/run/office2john.py) tool to extract the proper hash:
+
+![office2john.gif](/static/images/wolvctf-2023/office2john.gif)
```

Comment:
Pure textual information like shell transcript should be replaced by or accompanied by a textual representation.
Accompanied textual representation could be added as a multiline alt text or collapsed `<details>` section.